### DISCUSSION TITLE
Does 3blue1brown's quarternion video use this?

### DISCUSSION THREAD

**Discussion by tch1001** [manim-web/manim-web#15] [General]
I know 3blue1brown employed ben eater's help for his interactive quarternion video. But I can't find any information on how it is done, or if ben eater's code is open source. Any discussion on these? Would love to contribute to this project too! How may I help? Perhaps you can open some issues for me to tackle. :)

**Comment by hugo-s29** [comment]
Hi,

As far as I am aware, the *quaternion interactive experience*'s code is not open-source. In one of the comments, Ben Eater described how it works in more detail (http://disq.us/p/1wv9zek): these animations were built in javascript with `three.js` and `react`.

I am glad you want to help. Yes, I have not made issues to point out how to help with ManimWeb's development, but there have been some suggestions in the Discord server.

For example, creating a "plug-and-play" Mobject that can display digits. Currently, to render text, every `Tex` Mobject needs to run $\mathrm\LaTeX$ to display the required text. As you may guess, that's not really efficient: if you show a counter, like the one on 3Blue1Brown's counting collusion video, you shouldn't call $\mathrm\LaTeX$ 30,000 times. When displaying numbers, it's a lot simpler, you don't need $\mathrm\LaTeX$ to handle the whole number directly; you only need to know what's the shape of every digit, and then you can render them side-by-side to recreate the number to be displayed. I don't think this Mobject should be too tricky to implement.

Otherwise, you can help by adding documentation to the existing code. There is a special syntax (described in https://dart.dev/guides/language/language-tour#documentation-comments) that allows `dartdoc` to create the API reference (https://pub.dev/documentation/manim_web/latest/). For instance, the `AbstractDisplay` class has a lot more documentation than other classes.

Let me know (here or on Discord) if you need any help or more details.
